pkg/controller/common/helpers.go

```diff
@@ -236,13 +236,53 @@ func ConvertRawExtIgnitionToV2(inRawExtIgn *runtime.RawExtension) (runtime.RawEx
 	return outRawExt, nil
 }
 
+// Ensures SSH keys are unique for a given Ign 2 PasswdUser
+func dedupePasswdUserSSHKeys(passwdUser ign2types.PasswdUser) ign2types.PasswdUser {
+	// Map for checking for duplicates.
+	knownSSHKeys := map[ign2types.SSHAuthorizedKey]bool{}
+
+	// Preserve ordering of SSH keys.
+	dedupedSSHKeys := []ign2types.SSHAuthorizedKey{}
+
+	for _, sshKey := range passwdUser.SSHAuthorizedKeys {
+		if _, isKnown := knownSSHKeys[sshKey]; isKnown {
+			// We've seen this key before warn and move on.
+			glog.Warningf("duplicate SSH public key found: %s", sshKey)
+			continue
+		}
+
+		// We haven't seen this key before, add it.
+		dedupedSSHKeys = append(dedupedSSHKeys, sshKey)
+		knownSSHKeys[sshKey] = true
+	}
+
+	// Overwrite the keys with the deduped list.
+	passwdUser.SSHAuthorizedKeys = dedupedSSHKeys
+
+	return passwdUser
+}
+
+// Ensures SSH keys are unique for all Ign 2 PasswdUsers in a slice
+// See: https://bugzilla.redhat.com/show_bug.cgi?id=1934176
+func dedupeSSHKeys(passwdUsers []ign2types.PasswdUser) []ign2types.PasswdUser {
+	// Note to reviewers: Should we ignore any PasswdUsers that are not "core"?
+	for i := range passwdUsers {
+		passwdUsers[i] = dedupePasswdUserSSHKeys(passwdUsers[i])
+	}
+
+	return passwdUsers
+}
+
 // convertIgnition2to3 takes an ignition spec v2.2 config and returns a v3.2 config
 func convertIgnition2to3(ign2config ign2types.Config) (ign3types.Config, error) {
 	// only support writing to root file system
 	fsMap := map[string]string{
 		"root": "/",
 	}
 
+	// Ensure SSH keys are unique for each PasswdUser
+	ign2config.Passwd.Users = dedupeSSHKeys(ign2config.Passwd.Users)
```

Comment:
I think we can remove this, since basically every time this function gets called, the `removeIgnDuplicateFilesUnitsUsers` will also have been called.

Reply:
I noticed in [ConvertRawExtIgnitionToV3](https://github.com/openshift/machine-config-operator/blob/870f8c4de5a3927ef71e0cfd0e81bd5b75b424fd/pkg/controller/common/helpers.go#L152), `convertIgnition2to3` gets called without `removeIgnDuplicateFilesUnitsUsers` being called first.

However, based upon my testing, adding the deduplication call to `removeIgnDuplicateFilesUnitsUsers` is sufficient.